lib/cluster.js

```diff
@@ -260,6 +260,9 @@ function masterInit() {
       return process.nextTick(setupSettingsNT, settings);
     initialized = true;
     schedulingPolicy = cluster.schedulingPolicy;  // Freeze policy.
+    if (settings.scheduler) {
+      settings.scheduler(handles);
```

Comment:
The client provides the cb function and uses events to remove or add workers form the free property based on its own scheduling algorithm.